plugin/documents.py

```diff
@@ -419,11 +421,22 @@ def _clear_highlight_regions(self) -> None:
         for kind in global_settings.document_highlight_scopes.keys():
             self.view.erase_regions("lsp_highlight_{}".format(kind))
 
+    def is_in_higlighted_region(self, point: int) -> bool:
+        for kind in global_settings.document_highlight_scopes.keys():
+            regions = self.view.get_regions("lsp_highlight_{}".format(kind))
+            for r in regions:
+                if r.contains(point):
+                    return True
+        return False
+
     def _do_highlights(self) -> None:
-        self._clear_highlight_regions()
         if len(self.view.sel()) != 1:
             return
```

Comment:
Should we also clear highlight in case sel != 1 to match previous behavior?

Reply:
I believe that check exist so the line `point = self.view.sel()[0].begin()` don't throw an Index Error when trying to access the first item but the `view.sel()` returned an empty list...

I think that the condition should just be:
```python
    if not len(self.view.sel()):
            return
   point = self.view.sel()[0].begin()
```

Reply:
But it doesn't check that `len == 0`. I think it's supposed to also handle multiple cursors in which case we don't want to highlight matches.